### ISSUE TITLE
False negative: `redundant-unittest-assert` misses `assertEqual` / `assertNotEqual` between two constants

### ISSUE THREAD

**Issue by Semih702** [PyCQA/pylint#11321]
### Bug description

`redundant-unittest-assert` (W1503) flags `assertTrue` / `assertFalse` when the argument
is a literal, but not `assertEqual` / `assertNotEqual` when both operands are literals.

```python
import unittest


class ExampleTest(unittest.TestCase):
    def test_example(self):
        self.assertTrue("foo")  # flagged
        self.assertEqual(5, 5)  # not flagged
```

### Command used

```shell
pylint --disable=all --enable=redundant-unittest-assert example.py
```

### Pylint output

Only the `assertTrue` line is reported, nothing for `assertEqual(5, 5)`.

### Expected behavior

`assertEqual(5, 5)` reported as well.

W1503 does not evaluate anything today — `assertTrue(True)` and `assertTrue(0)` are both
flagged — so the rule is "the outcome of this assertion is fixed by the source" rather
than "this assertion always passes". Two literal operands fit that same rule, which is why
this looks like W1503 rather than a new message.

A narrow scope would be enough: only `assertEqual` / `assertNotEqual`, and only when both
of the first two positional arguments are literals. That leaves out containers, folded
expressions such as `1 + 1`, and the keyword form. `assertIs` should stay out, since
identity between equal literals is implementation defined.

### Root cause

`pylint/checkers/stdlib.py`, `StdlibChecker._check_redundant_assert`: the name set is
`{"assertTrue", "assertFalse"}`, and only `node.args[0]` is examined.

### Pylint version

```shell
pylint 4.1.0-dev0 (main)
astroid 4.2.0b5
Python 3.12.10
```

I can open a PR if this sounds reasonable.
